rustworkx-core/src/coloring.rs

```diff
@@ -164,44 +368,61 @@ where
 /// use petgraph::graph::NodeIndex;
 /// use petgraph::Undirected;
 /// use rustworkx_core::dictmap::*;
-/// use rustworkx_core::coloring::greedy_node_color;
+/// use rustworkx_core::coloring::{greedy_node_color, GreedyStrategyCore};
 ///
 /// let g = Graph::<(), (), Undirected>::from_edges(&[(0, 1), (0, 2)]);
-/// let colors = greedy_node_color(&g);
+/// let colors = greedy_node_color(&g, GreedyStrategyCore::Degree);
 /// let mut expected_colors = DictMap::new();
 /// expected_colors.insert(NodeIndex::new(0), 0);
 /// expected_colors.insert(NodeIndex::new(1), 1);
 /// expected_colors.insert(NodeIndex::new(2), 1);
 /// assert_eq!(colors, expected_colors);
 /// ```
 ///
-pub fn greedy_node_color<G>(graph: G) -> DictMap<G::NodeId, usize>
+pub fn greedy_node_color<G>(
+    graph: G,
+    greedy_strategy: GreedyStrategyCore,
```

Comment:
I worry about the API implications of this, wouldn't this be a breaking API change because you're required to pass a strategy enum when calling this function now?

Reply:
As we have discussed, the existing functions now remain unscathed, and two new functions `greedy_node_color_with_coloring_strategy` and `greedy_edge_color_with_coloring_strategy` have been added. I have also changed their return type to be `Result<Dict<...>, E>` to allow intercepting errors when using the preset function.